source/mesh_deformation/interface.cc

```diff
@@ -664,6 +877,11 @@ namespace aspect
 
       // We can safely close this now
       mesh_vertex_constraints.close();
+
+      // if we are just starting, we need to initialize the mesh displacement vector.
+      if (this->simulator_is_past_initialization() == false ||
```

Comment:
Hi @gassmoeller, I'm just curious what your thoughts are on this :)

Reply:
I think we need to run this several times in case of initial adaptive refinement at the boundary (each time the deformation is initialized with 0), because we want the final deformation for the final resolution to conform as closely as possible to the prescribed initial topography. Do you think there are situations where this would be wrong?

Reply:
No that makes total sense. Don't know why I was doubting, maybe I thought that it could in principle be done once at the end of initial mesh refinement unless you wanted to refine based on topography. Anyway, this is good to go from my side as well!